src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -155,25 +152,66 @@ public override TPredictor Train(TrainContext context)
 
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
-        public SymSgdClassificationTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="SymSgdClassificationTrainer"/>
+        /// </summary>
+        /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name for the column containing the initial weight.</param>
```

Comment:
>param name="weightColumn">The name for the column containing the initial weight.</param> [](start = 13, length = 88)

If we don't support weighted training this should be removed, at least in its description of a column.

I *think* what was actually intended, because it says "initial weight," was that when you train, an initial predictor from which you get initial weights should be supported. (e.g., as we see here:)

[https://github.com/dotnet/machinelearning/blob/893a3852bded9b8785b39e7973c02758069621b8/src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs#L145](https://github.com/dotnet/machinelearning/blob/893a3852bded9b8785b39e7973c02758069621b8/src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs#L145)

However this would be a parameter to the training/fitting method, not the constructor.

Reply:
That is actually already taken care of.
TrainerEstimatorBase has TrainTransformer method, which takes an initial predictor, and passes that to TrainModelCore through the trainContext

---
In reply to: [220342250](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220342250) [](ancestors = 220342250,220340515)

Reply:
You're talking about this I guess.

[https://github.com/dotnet/machinelearning/blob/d7b062de20c091cb4af5f58dda4bd0b98ad66c3e/src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs#L127-L128](https://github.com/dotnet/machinelearning/blob/d7b062de20c091cb4af5f58dda4bd0b98ad66c3e/src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs#L127-L128)

@artidoro, no, as you see this method is *protected*. It is not accessible to people that instantiate the class. So, no, not taken care of.

Your PR was already merged, so whoops, too late I guess. Maybe we can file an issue to fix this post 0.6.

---
In reply to: [220345387](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220345387) [](ancestors = 220345387,220342250,220340515)